### DISCUSSION TITLE
Remove the ready-to-use agent

### DISCUSSION THREAD

**Discussion by peterpeterparker** [dfinity/sdk#2761] [Feature Requests]
I would like to remove the ready-to-use agent for the canister that is generated automatically in the `index.js` of the `declarations`.

It is an issue in pre-rendered projects such as SvelteKit (`Must specify a host to connect to. Error: Must specify a host to connect to.`) and when used in NodeJS scripts as well.

e.g.

```
/**
 * A ready-to-use agent for the backend canister
 * @type {import("@dfinity/agent").ActorSubclass<import("./backend.did.js")._SERVICE>}
 */
export const backend = createActor(canisterId);
```

**Comment by krpeacock** [comment]
`backend -> declarations -> node_compatibility: true` will do that for ya

**Reply by peterpeterparker** [reply]
Cool!
